sdk/storage/azure-storage-blob/tests/_shared/testcase.py

```diff
@@ -353,27 +373,102 @@ def storage_account():
+
     try:
-        rg = rg_preparer.create_resource(rg_name)
-        StorageTestCase._RESOURCE_GROUP = rg['resource_group']
-        try:
-            storage_dict = storage_preparer.create_resource(
-                storage_name,
-                resource_group=rg['resource_group']
+        if i_need_to_create_rg:
+            rg_name = create_random_name("pystorage", 24)
+            rg = rg_preparer.create_resource(rg_name, location=location)['resource_group']
+        else:
+            rg_name = existing_rg_name or "no_rg_needed"
+            rg = FakeResource(
+                name=rg_name,
+                id="/subscriptions/{}/resourceGroups/{}".format(subscription_id, rg_name)
             )
-            # Now the magic begins
-            StorageTestCase._STORAGE_ACCOUNT = storage_dict['storage_account']
-            StorageTestCase._STORAGE_KEY = storage_dict['storage_account_key']
+        StorageTestCase._RESOURCE_GROUP = rg
+
+        try:
+            if got_storage_info_from_env:
+
+                if storage_connection_string:
+                    storage_connection_string_parts = dict([
+                        part.split('=', 1)
+                        for part in storage_connection_string.split(";")
+                    ])
+
+                storage_account = None
+                if existing_storage_name:
+                    storage_name = existing_storage_name
+                    storage_account = StorageAccount(
+                        location=location,
+                    )
+                    storage_account.name = storage_name
+                    storage_account.id = storage_name
+                    storage_account.primary_endpoints=Endpoints()
+                    storage_account.primary_endpoints.blob = 'https://{}.{}.core.windows.net'.format(storage_name, 'blob')
+                    storage_account.primary_endpoints.queue = 'https://{}.{}.core.windows.net'.format(storage_name, 'queue')
+                    storage_account.primary_endpoints.table = 'https://{}.{}.core.windows.net'.format(storage_name, 'table')
+                    storage_account.primary_endpoints.file = 'https://{}.{}.core.windows.net'.format(storage_name, 'file')
+                    storage_key = existing_storage_key
+
+                if not storage_connection_string:
+                    storage_connection_string=";".join([
+                        "DefaultEndpointsProtocol=https",
+                        "AccountName={}".format(name),
+                        "AccountKey={}".format(storage_key),
+                        "BlobEndpoint={}".format(storage_account.primary_endpoints.blob),
+                        "TableEndpoint={}".format(storage_account.primary_endpoints.table),
+                        "QueueEndpoint={}".format(storage_account.primary_endpoints.queue),
+                        "FileEndpoint={}".format(storage_account.primary_endpoints.file),
+                    ])
+
+                if not storage_account:
```

Comment:
>f not storage_account: [](start = 17, length = 22)

Probably we want connection string to have a higher priority than existing_storage_name.
eg. for government cloud, even there is a storage account name provided, we shouldn't help them construct primary endpoints because they always use the customized ones.

Reply:
There is actually no concept of priority here:
- if you provide only a storage account name, I re-build a connection string.
- If you provide only a connection string, I re-build a ARM resource JSON
- If you provide both, I re-build nothing
And then I pass both to the test, that decides what he wants to do with it. We could do a client builder for the tests that says "if both, then use connection string, if not both, use the one you got", but that won't be in htat part of the code